### DISCUSSION TITLE
Supabase Row level Security CHECK

### DISCUSSION THREAD

**Discussion by Othman-Sami** [supabase/supabase#21504] [Questions] [answered]
Hi 
I want to ask  if I can make Supabase Row level Security CHECK 
for table 
like just specific user (email) can access all CRUD
and others just read

**Comment by GaryAustin1** [comment]
Yes.
You could by doing insert/delete/update policies for authenticated role and ` auth.jwt()->>'email' = 'myemail@myorg.com'`
And then select to policy to roles you want with `true` policy.

**Reply by Othman-Sami** [reply]
can you explain in code 

I mean like this??


`CREATE POLICY "control user" ON "public"."user"`
`AS PERMISSIVE FOR ALL`
`TO authenticated`
`USING (auth.jwt()->>'email' = 'myemail@myorg.com')`

**Reply by GaryAustin1** [reply]
You need an additional Select policy with USING and true policy if you want anyone to read.

**Reply by Othman-Sami** [reply]
If there is no disturbance
Can you help me write the code?

**Reply by GaryAustin1** [reply]
Your policy looked fine for the email part.
You just need:
```
CREATE POLICY "allow select" ON "public"."user"
AS PERMISSIVE FOR SELECT
TO authenticated
USING (true);
```
Assuming you only want authenticated users to read.

Not tested, I'm just pointing you in the hopefully right direction.

**Reply by Othman-Sami** [reply]
the code for email not work , any authenticated email can update or delete

My Idea is to make like admin email for edit in table am I in right way or there is another way ??

**Reply by GaryAustin1** [reply]
If you have RLS enabled and have this policy: 
```
CREATE POLICY "control user" ON "public"."user"
AS PERMISSIVE FOR ALL
TO authenticated
USING (auth.jwt()->>'email' = 'myemail@myorg.com')
```
And only 1 other policy for select.  Then only the user with that email should be able to update.

**Reply by Othman-Sami** [reply, marked as answer]
after many tries  that work for me 

```
CREATE POLICY "control user" ON "public"."user"
AS PERMISSIVE FOR ALL
TO authenticated
USING (auth.jwt()->>'email' = 'myemail@myorg.com')
WITH CHECK (auth.jwt()->>'email' = 'myemail@myorg.com');
```
